traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java

```diff
@@ -178,42 +175,42 @@ public static AuthenticatedUrlRequest orders(OrderQuery query, Session session,
         if (query.getFromDate() != null)
             rb.param("from", DateHelper.formatDateToISO8601(query.getFromDate()));
 
-        return rb.build(session.getToken());
+        return rb.buildAuthenticatedUrlRequest();
     }
 
     public static AuthenticatedUrlRequest cancellationQuote(Order order, Session session, Context context) {
-        return new RouteBuilder(context, session.getApiKey())
+        return new RouteBuilder(context, session)
                 .method(NetworkTask.Route.Method.GET)
                 .path("/order/" + order.getId() + "/cancellation")
-                .build(session.getToken());
+                .buildAuthenticatedUrlRequest();
     }
 
     public static AuthenticatedUrlRequest cancelOrder(CancellationQuote quote, Session session, Context context) {
-        return new RouteBuilder(context, session.getApiKey())
+        return new RouteBuilder(context, session)
                 .method(NetworkTask.Route.Method.PATCH)
                 .path("/order/" + quote.getOrder().getId() + "/cancellation/" + quote.getId())
-                .build(session.getToken());
+                .buildAuthenticatedUrlRequest();
     }
 
     public static AuthenticatedUrlRequest emailOrderConfirmation(Order order, Session session, Context context) {
-        return new RouteBuilder(context, session.getApiKey())
+        return new RouteBuilder(context, session)
                 .method(NetworkTask.Route.Method.PATCH)
                 .path("/order/" + order.getId() + "/ticket")
-                .build(session.getToken());
+                .buildAuthenticatedUrlRequest();
     }
 
     private static class RouteBuilder {
         private NetworkTask.Route.Method method;
         private String path;
-        private String url;
         private JSONPayloadProvider payload = null;
-        private String apiKey;
         private Map<String, String> headers;
         private List<Pair<String, String>> params;
-        private Context context;
+        private Session session;
+        private String travelerSDKEndpoint;
 
-        private RouteBuilder(Context context, String apiKey) {
+        private RouteBuilder(Context context, Session session) {
```

Comment:
Im struggling to see the value of having this class have a reference to `Session` or even know of it.

Best if it doesn't and is model agnostic. Maybe I missed sth, but I don't see any use other than token here, which is best passed as an argument to the build method.

What happens now is that this class is making an assumption that session will store token and the token is valid for the duration of the session. While that may be true it is not within this classes responsibility scope to be aware of .

Reply:
Given that if router doesn't have knowledge of session, session's fields will have to be passed in piecemeal as needed.